### DISCUSSION TITLE
can i get the depthImage directly in icub 2.7

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#662] [Q&A] [answered]
Hi,

icub 2.7 is equipped with stereo camera, can i directly get the depthImage or disparity map from the output port as icubSim did in gazebo like below.
```
    <connection>
        <from>/icubSim/cam/left/depthImage:o</from>
        <to>/icub-grasp/depth:i</to>
        <protocol>fast_tcp</protocol>
    </connection>
```

Thanks.

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

To obtain depth images from the stereo rig of the real robot, you ought to perform [structure from motion][1].
To this end, you can explore the codebase hosted at https://github.com/robotology/stereo-vision.

However, the moving eye bulbs of iCub pose difficulties in getting decent depth images with the required levels of detail that could enable successful reaching. 
For this, you may consider relying on an external RealSense device that can be plugged on a custom holder.
See the relative upgrade kit at:
- https://icub-tech-iit.github.io/documentation/upgrade_kits/realsense_holder/support/

We provide all the info to [print the holder yourself][2] and then calibrate it.

[1]: https://en.wikipedia.org/wiki/Structure_from_motion
[2]: https://icub-tech-iit.github.io/documentation/upgrade_kits/realsense_holder/support/#print-the-realsense-holder-yourself
